pkg/snapshotter/loopdevice.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright © 2022 - 2023 SUSE LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package snapshotter
+
+import (
+	"path/filepath"
+
+	"github.com/rancher/elemental-toolkit/pkg/constants"
+
+	v1 "github.com/rancher/elemental-toolkit/pkg/types/v1"
+	"github.com/rancher/elemental-toolkit/pkg/utils"
+)
+
+const (
+	loopDeviceSnapsPath    = ".snapshots"
+	loopDeviceImgName      = "snapshot.img"
+	loopDeviceWorkDir      = "snapshot.workDir"
+	loopDeviceLabelPattern = "EL_SNAP%d"
+	loopDevicePassiveSnaps = loopDeviceSnapsPath + "/passives"
+)
+
+var _ v1.Snapshotter = (*LoopDevice)(nil)
+
+type LoopDevice struct {
+	cfg            v1.Config
+	snapshotterCfg v1.SnapshotterConfig
+	loopDevCfg     v1.LoopDeviceConfig
+	rootDir        string
+	/*currentSnapshotID int
```

Comment:
Should these fields be kept?

Reply:
They are commented because the linter complained about being unused 🤷🏽‍♂️ They are back and being used on on the next PR I am preparing with the full implemenentation of this `loopdevice.go`, currently writing unit tests.